include/swift/AST/DeclContext.h

```diff
@@ -255,6 +256,10 @@ class alignas(1 << DeclContextAlignInBits) DeclContext {
   /// EnumDecl, otherwise return null.
   EnumDecl *getAsEnumOrEnumExtensionContext() const;
 
+  /// If this DeclContext is a struct, or an extension on a struct, return the
+  /// StructDecl, otherwise return null.
+  StructDecl *getAsStructOrStructExtensionContext() const;
```

Comment:
You might want to do a pass over callers of getAsNominalTypeOrNominalTypeExtensionContext() and see if any just `dyn_cast<StructDecl>` the result and clean them up.

Reply:
I found 98 instances of `getAsNominalTypeOrNominalTypeExtensionContext`, but none of them appear to unconditionally cast to `StructDecl` afterward.